hphp/runtime/base/variable-serializer.cpp

```diff
@@ -240,12 +240,15 @@ void VariableSerializer::write(double v) {
     if (!std::isinf(v) && !std::isnan(v)) {
       char *buf;
       vspprintf(&buf, 0, "%.*k", precision, v);
+      if (m_option & k_JSON_PRESERVE_ZERO_FRACTION
+          && strchr(buf, '.') == nullptr) {
+        free(buf);
+        asprintf(&buf, "%.1f", v);
```

Comment:
I think precision only truncates and does not add trailing zeros.
However, you are correct about the locale. I didn't think of it. Thanks!